sched/sched/sched_get_stackinfo.c

```diff
@@ -101,5 +111,7 @@ int nxsched_get_stackinfo(pid_t pid, FAR struct stackinfo_s *stackinfo)
   stackinfo->adj_stack_size  = qtcb->adj_stack_size;
   stackinfo->stack_alloc_ptr = qtcb->stack_alloc_ptr;
   stackinfo->adj_stack_ptr   = qtcb->adj_stack_ptr;
+  stackinfo->tls_info_ptr    = qtcb->tls_info_ptr;
```

Comment:
This is not good.  Other than initial setup, TLS is 100% a user-space function.  It must be implemented entirely in libc (the system call to get the stack base is a KLUDGE to avoid aligning stacks).  But moving the TLS pointer into the TCB is just wrong.  I cannot merge this kind of change.

Reply:
> This is not good. Other than initial setup, TLS is 100% a user-space function. It must be implemented entirely in libc (the system call to get the stack base is a KLUDGE to avoid aligning stacks). But moving the TLS pointer into the TCB is just wrong. I cannot merge this kind of change.

I save this poiner to fix the wrong assumption(push-down stack) you made in #987:
https://github.com/apache/incubator-nuttx/commit/bda24f09c2fe0d72b3452686052d075fa7f416ae#diff-dd1838d4b8adff099b4fbcd041c8ec8e7f70cdf1fb9e360781b7298352105be6R68-R72

Two alternative select:

1. Kee the push-down stack assumption as before, and remove tls_info_ptr
2. Guard tls_info_ptr by CONFIG_TLS_ALIGNED since tls_info_ptr isn't really used in this case.

Which one do you prefer?

Reply:
> I save this poiner to fix the wrong assumption(push-down stack) you made in #987:
> [bda24f0#diff-dd1838d4b8adff099b4fbcd041c8ec8e7f70cdf1fb9e360781b7298352105be6R68-R72](https://github.com/apache/incubator-nuttx/commit/bda24f09c2fe0d72b3452686052d075fa7f416ae#diff-dd1838d4b8adff099b4fbcd041c8ec8e7f70cdf1fb9e360781b7298352105be6R68-R72)

This is not a wrong assumption and has nothing to do with a push-down stack.  It is correct logic.  You logic is wrong.  The TLS data must lie at the lowest address of the memory allocation or it cannot be accessed in user space use via:

tlsinfo = (FAR struct tls_info_s *)((builtin_sp()) & TLS_MASK).

That is necessary for pure user-space access.  You changes are not good and should not be merged (although it is difficult to understand what all is going on in 207 files of mixed changes).

Reply:
Ok, if TLS is always put at the lowest address regardless "push down stack" or "push up stack". The code is right, we just need fix the comment.